bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -567,4 +582,27 @@ private boolean isAddonInstalled(String addonId) {
         }
         return false;
     }
+
+    /**
+     * Get mac address associated with given Internet socket address
+     * 
+     * @param inetSocketAddress the Internet address
+     * @return the mac address in colon delimited upper-case hex e.g. '01:02:03:04:A5:B6:C7:D8'
+     * @throws SocketException if address is not on this PC, or no mac address is associated
+     */
+    private String macAddressFrom(InetSocketAddress inetSocketAddress) throws SocketException {
+        NetworkInterface networkInterface = NetworkInterface.getByInetAddress(inetSocketAddress.getAddress());
+        if (networkInterface == null) {
+            throw new SocketException("No network interface");
+        }
+        byte[] macBytes = networkInterface.getHardwareAddress();
+        if (macBytes == null) {
+            throw new SocketException("No mac address");
+        }
+        StringBuilder resultBuilder = new StringBuilder();
+        for (byte macByte : macBytes) {
+            resultBuilder.append(String.format("%02X:", macByte));
```

Comment:
what are your thoughts on specifying different formats? For example, WiZ doesn't include colons

Reply:
I was thinking about this :)

There are two possible request formats -- namely 'requestPlain' and 'request'. So I just uploaded a new commit whereby if 'requestPlain' is used, it uses colon delimited upper-case hex, or if 'request' is used it uses plain bytes.

However I have a horrible feeling that we may indeed need to support a format parameter token whereby 'requestPlain' can use delimiters ["", "-", ":"] and upper/lower case hex.

Reply:
So I just added another token `fmtMac` which allows one to provide a format specifier..